openfisca_core/tools/test_runner.py

```diff
@@ -131,38 +135,51 @@ def parse_test(self):
 
         self.tax_benefit_system = _get_tax_benefit_system(self.baseline_tax_benefit_system, self.test.get('reforms', []), self.test.get('extensions', []))
 
+        builder = SimulationBuilder()
+        input = self.test.get('input', {})
+        period = self.test.get('period')
+        verbose = self.options.get('verbose')
+        performance = self.options.get('performance')
+
         try:
-            builder = SimulationBuilder()
-            input = self.test.get('input', {})
-            period = self.test.get('period')
-            verbose = self.options.get('verbose')
             builder.set_default_period(period)
             self.simulation = builder.build_from_dict(self.tax_benefit_system, input)
-            self.simulation.trace = verbose
         except (VariableNotFound, SituationParsingError):
             raise
         except Exception as e:
             error_message = os.linesep.join([str(e), '', f"Unexpected error raised while parsing '{self.fspath}'"])
             raise ValueError(error_message).with_traceback(sys.exc_info()[2]) from e  # Keep the stack trace from the root error
 
-    def runtest(self):
-        self.parse_test()
-        verbose = self.options.get('verbose')
         try:
+            self.simulation.trace = verbose or performance
             self.check_output()
         finally:
+            tracer = self.simulation.tracer
             if verbose:
-                print("Computation log:")  # noqa T001
-                self.simulation.tracer.print_computation_log()
+                self.print_computation_log(tracer)
+            if performance:
+                self.create_performance_log(tracer)
+
+    def print_computation_log(self, tracer):
+        print("Computation log:")  # noqa T001
+        tracer.print_computation_log()
+
+    def create_performance_log(self, tracer):
+        f = open("performance.json", "w")
```

Comment:
This could be wrapped in a `tracer.performance_log.generate_graph(dir_path)` method. 
That would allow to generate the graph in another context than a text execution (e.g. a calculation on a large population), and I think would better respect the Single Responsibility Principle.